cdap-master/src/main/java/co/cask/cdap/data/tools/QueueConfigUpgrader.java

```diff
@@ -61,7 +61,13 @@ protected TableId getTableId() {
 
   @Nullable
   private QueueName fromRowKey(byte[] oldRowKey) {
-    QueueName queueName = QueueName.from(oldRowKey);
+    QueueName queueName;
+    try {
+      queueName = QueueName.from(oldRowKey);
```

Comment:
Ideally the actual rewriting of data (scan and copy) happens in one place (QueueAdmin.upgrade()) so that all knowledge of the upgrade is in one place.